cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -566,14 +568,39 @@ impl ManagedEnvironment {
                             _ => ManagedEnvironmentError::Git(err),
                         })?;
                 }
-                // if it still doesn't exist after fetching, error
-                if !floxmeta
+                // If it still doesn't exist after fetching,
+                // the upstream branch has diverged from the local branch.
+                let in_remote = floxmeta
                     .git
                     .branch_contains_commit(&lock.rev, &remote_branch)
-                    .map_err(ManagedEnvironmentError::Git)?
-                {
+                    .map_err(ManagedEnvironmentError::Git)?;
+
+                if in_remote {
+                    return Ok(lock);
+                }
+
+                // locked reference not found in remote/sync branch
+                // check if it's in the project's branch.
+                // If the project's branch doesn't exist, or the project was moved,
+                // this will still fail to resolve.
+
+                let local_branch = branch_name(&flox.system, pointer, dot_flox_path);
+
+                let has_branch = floxmeta
+                    .git
+                    .has_branch(&local_branch)
+                    .map_err(ManagedEnvironmentError::CheckBranchExists)?;
+
+                let in_local = has_branch
+                    && floxmeta
+                        .git
+                        .branch_contains_commit(&lock.rev, &local_branch)
+                        .map_err(ManagedEnvironmentError::Git)?;
+
+                if !in_remote && !in_local {
```

Comment:
```suggestion
                if !in_local {
```

Reply:
yeah !in_remote is implied but i think its good for clarity (i'd otherwise add a comment saying the same)